lib/HALTest/src/MqttClient.h

```diff
@@ -0,0 +1,149 @@
+ * Types and Classes
+ *****************************************************************************/
+
+/** This class provides mqtt client functionality. */
+class MqttClient : public IMqttClient
+{
+public:
+    /**
+     * Constructs the MQTTClient.
+     */
+    MqttClient();
+
+    /**
+     * Destroys the MQTTClient.
+     */
+    virtual ~MqttClient();
+
+    /**
+     * Initialize MQTTClient state.
+     *
+     * @return If successfully initialized, returns true. Otherwise, false.
+     */
+    bool init() final;
+
+    /**
+     * Process communication with the broker.
+     *
+     * @return If communication is successful, returns true. Otherwise, false.
+     */
+    bool process() final;
+
+    /**
+     * Set client configuration.
+     *
+     * @param[in] settings Mqtt settings struct.
+     * @return If successfully set, returns true. Otherwise, false.
+     */
+    bool setConfig(const MqttSettings& settings) final;
+
+    /**
+     * Start connection to the broker.
+     * This method does not necessarily wait for the connection to be established, it just starts the connection
+     * process. Check `isConnected()` for the current connection status.
+     *
+     * @return If connection has been succesfully started, returns true. Otherwise, false.
+     */
+    bool connect() final;
+
+    /**
+     * Disconnect from the broker.
+     */
+    void disconnect() final;
+
+    /**
+     * Is connected to the broker?
+     *
+     * @return If connected, it will return true otherwise false.
+     */
+    bool isConnected() const final;
+
+    /**
+     * Publishes a message to the broker.
+     *
+     * @param[in] topic                     Topic to publish to.
+     * @param[in] useClientIdAsBaseTopic    If true, the client ID is used as the base (prefix) of the topic.
+     * @param[in] message                   Message to publish.
+     *
+     * @return If successfully published, returns true. Otherwise, false.
+     */
+    bool publish(const String& topic, const bool useClientIdAsBaseTopic, const String& message) final;
+
+    /**
+     * Subscribes to a topic.
+     *
+     * @param[in] topic                     Topic to subscribe to.
+     * @param[in] useClientIdAsBaseTopic    If true, the client ID is used as the base (prefix) of the topic.
+     * @param[in] callback                  Callback function, which is called on a new message.
+     *
+     * @return If successfully subscribed, returns true. Otherwise, false.
+     */
+    bool subscribe(const String& topic, const bool useClientIdAsBaseTopic, TopicCallback callback) final;
+
+    /**
+     * Unsubscribes from a topic.
+     *
+     * @param[in] topic                     Topic to unsubscribe from.
+     * @param[in] useClientIdAsBaseTopic    If true, the client ID is used as the base (prefix) of the topic.
+     */
+    void unsubscribe(const String& topic, const bool useClientIdAsBaseTopic) final;
+};

```

Comment:
Can the original one be copied too?

Reply:
The MQTTClient of Sim? Or which original do you mean?

Reply:
The one you address with this stubbed one.

Reply:
Sorry I dont understand what you mean